src/Utils.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.0 <0.9.0;
+
+contract Utils {
+    address internal constant CONSOLE2_ADDRESS = address(0x000000000000000000636F6e736F6c652e6c6f67);
+    uint256 internal constant UINT256_MAX =
+        115792089237316195423570985008687907853269984665640564039457584007913129639935;
+
+    function bound(uint256 x, uint256 min, uint256 max) internal virtual returns (uint256 result) {
+        require(min <= max, "Test bound(uint256,uint256,uint256): Max is less than min.");
+
+        uint256 size = max - min;
+
+        if (size == 0) {
+            result = min;
+        } else if (size == UINT256_MAX) {
+            result = x;
+        } else {
+            ++size; // make `max` inclusive
+            uint256 mod = x % size;
+            result = min + mod;
+        }
+        
+        // Log the result with console2.log if supported
+        (bool status, ) = address(CONSOLE2_ADDRESS).staticcall(
+            abi.encodeWithSignature("log(string,uint)", "Bound result", result)
+        );
+        // Silence compiler warning
+        status; 
+    }
+
+    /// @dev Compute the address a contract will be deployed at for a given deployer address and nonce
+    /// @notice adapated from Solmate implementation (https://github.com/Rari-Capital/solmate/blob/main/src/utils/LibRLP.sol)
+    function computeCreateAddress(address deployer, uint256 nonce) internal pure virtual returns (address) {
```

Comment:
Thoughts on throwing in a `computeCreate2Address` method in this PR?

```solidity
function computeCreate2Address(bytes32 salt, bytes32 initcodeHash, address deployer) internal pure returns (address) {
    return addressFromLast20Bytes(keccak256(abi.encodePacked(bytes1(0xff), deployer, salt, initcodeHash)));
}
```

Reply:
Well, we've gone so far already. Why not? 😅